### PR TITLE
feat(error): an idea for simple stack

### PR DESCRIPTION
**Pull request by JiaLiPassion**
An idea to only throw one error to detect most zone stack trace frames.

### CODE REVIEW

File: lib/zone.ts [hunk 1]
```diff
@@ -1834,17 +1830,60 @@ const Zone: ZoneType = (function(global: any) {
...
+            (t as any)._transitionTo = fakeTransitionTo;
+            t.invoke();
+          },
+          () => {});
     });
-  };
-  // Cause the error to extract the stack frames.
-  detectZone.runTask(detectZone.scheduleMacroTask('detect', detectRunFn, null, () => null, null));
+  });
+  Error.stackTraceLimit = 15;
```
Comment (mhevery) [inline lib/zone.ts]: we should store and restore, rather than reset to 15

File: lib/zone.ts [hunk 2]
```diff
@@ -1834,17 +1830,60 @@ const Zone: ZoneType = (function(global: any) {
...
+    }
+  });
+
+  Error.stackTraceLimit = 100;
+  childDetectZone.run(() => {
+    childDetectZone.runGuarded(() => {
+      const fakeTransitionTo =
+          (toState: TaskState, fromState1: TaskState, fromState2: TaskState) => {};
+      childDetectZone.scheduleEventTask(
+          'detect',
```
Comment (mhevery) [inline lib/zone.ts]: to keep the size down, can we reuse existing string such as `blacklistedStackFramesSymbol`?

File: lib/zone.ts (hunk 1 above)
Comment (JiaLiPassion) [inline lib/zone.ts]: Sure, will do.

File: lib/zone.ts (hunk 2 above)
Comment (JiaLiPassion) [inline lib/zone.ts]: Yes, you mean remove the duplicate frame from blacklistedstackFrames? I will modify the source and let you review.

Comment (JiaLiPassion) [inline lib/zone.ts]: Sorry, I am not sure I understand your meaning, could you explain more? thank you.

**Pull request merged**